api/src/filters.rs

```diff
@@ -2,21 +2,48 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{context::Context, handlers};
+use diem_api_types::Error;
 
 use std::convert::Infallible;
-use warp::{Filter, Rejection, Reply};
+use warp::{http::StatusCode, reply, Filter, Rejection, Reply};
 
-pub fn routes(context: Context) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
-    index(context)
+pub fn routes(context: Context) -> impl Filter<Extract = impl Reply, Error = Infallible> + Clone {
+    index(context.clone())
+        .or(get_account_resources(context))
+        .recover(handle_rejection)
 }
 
 // GET /
 pub fn index(context: Context) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
     warp::path::end()
+        .and(warp::get())
         .and(with_context(context))
         .and_then(handlers::index)
 }
 
+// GET /accounts/<address>/resources
+pub fn get_account_resources(
+    context: Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    warp::path!("accounts" / String / "resources")
+        .and(warp::get())
+        .and(with_context(context))
+        .and_then(handlers::get_account_resources)
```

Comment:
Let's make the `handlers::get_account_resource` function more type safe by pushing argument validation outside of the function and into the warp path constructor. I'm specifically talking about the `AccountAddress` argument which is being parsed out of the path as a `String` and then passed to the handler where there handler then immediately calls `Address::from_str(&address_str)`, instead we can replace `String` with `Address` in the path (ensuring that `Address` impls `FromStr`).

```
warp::path!("accounts" / Address / "resources")
```

Reply:
the problem with this is error response will be handled by warp, which is very simple and lost address information.

Reply:
be more specific, when Address parsing failed, warp returns NotFound error, I think it is better we return invalid address error instead.